LoopKitUI/View Controllers/InsulinSensitivityScheduleViewController.swift

```diff
@@ -0,0 +1,397 @@
+        if editingStyle == .delete {
+            internalItems.remove(at: indexPath.row)
+
+            super.tableView(tableView, commit: editingStyle, forRowAt: indexPath)
+
+            if internalItems.count == 1 {
+                self.isEditing = false
+            }
+
+            updateInsertButton()
+            updateEditButton()
+            updateTimeLimitsFor(itemAt: indexPath.row-1)
+            updateTimeLimitsFor(itemAt: indexPath.row)
+
+        }
+    }
+
+    open override func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+        if sourceIndexPath != destinationIndexPath {
+            let item = internalItems.remove(at: sourceIndexPath.row)
+            internalItems.insert(item, at: destinationIndexPath.row)
+
+            guard destinationIndexPath.row > 0, let cell = tableView.cellForRow(at: destinationIndexPath) as? SetConstrainedScheduleEntryTableViewCell else {
+                return
+            }
+
+            let interval = cell.minimumTimeInterval
+            let startTime = internalItems[destinationIndexPath.row - 1].startTime + interval
+
+            internalItems[destinationIndexPath.row] = RepeatingScheduleValue(startTime: startTime, value: internalItems[destinationIndexPath.row].value)
+
+            DispatchQueue.main.async {
+                tableView.reloadRows(at: [destinationIndexPath], with: .none)
+                self.updateTimeLimitsForItemsAdjacent(to: destinationIndexPath.row)
+            }
+        }
+    }
+
+    // MARK: - UITableViewDelegate
+
+    open override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
+        guard indexPath.section == 0 else {
+            return super.tableView(tableView, shouldHighlightRowAt: indexPath)
+        }
+
+        return !isReadOnly
+    }
+
+    open override func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
+        tableView.beginUpdates()
+        hideSetConstrainedScheduleEntryCells(excluding: indexPath)
+        tableView.endUpdates()
+        return super.tableView(tableView, willSelectRowAt: indexPath)
+    }
+
+    open override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        super.tableView(tableView, didSelectRowAt: indexPath)
+
+        switch Section(rawValue: indexPath.section)! {
+        case .schedule:
+            break
+        case .sync:
+            if let schedule = schedule {
+                insulinSensitivityScheduleStorageDelegate?.saveSchedule(schedule, for: self, completion: { (result) in
+                    switch result {
+                    case .success:
+                        self.delegate?.dailyValueScheduleTableViewControllerWillFinishUpdating(self)
+                        self.isScheduleModified = false
+                        self.updateInsertButton()
```

Comment:
Technically we'd update the cancel button here, too, I think.

Reply:
That gets updated via the isScheduleModified var setter